src/trusted-device/Security/TwoFactor/Trusted/JwtTokenEncoder.php

```diff
@@ -55,14 +54,18 @@ public function decodeToken(string $token): ?Plain
             return null;
         }
 
-        if (!$this->configuration->validator()->validate($token, ...$this->configuration->validationConstraints())) {
-            return null;
-        }
-
-        if ($token->isExpired(new DateTimeImmutable())) {
+        if (!$this->configuration->validator()->validate($token, ...$this->validationConstraints())) {
             return null;
         }
 
         return $token;
     }
+
+    /** @return iterable<int, Constraint> */
+    private function validationConstraints(): iterable
+    {
+        yield new Constraint\SignedWith($this->configuration->signer(), $this->configuration->signingKey());
+        yield new Constraint\ValidAt($this->clock); // replace with LooseValidAt once dependency on lcobucci/jwt is bumped up
+        yield from $this->configuration->validationConstraints();
```

Comment:
I kept this here but it's very unlikely to have users adding validation constraints to the injected configuration (it's a possibility though, and custom claims are configurable by providing a different builder factory).